deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/AutoPromoter.java

```diff
@@ -122,32 +123,116 @@ boolean isCurrentDeployRetirable(String envId, DeployBean currDeployBean) throws
         return false;
     }
 
-    void promoteBuild(EnvironBean currEnvBean, DeployBean currDeployBean, int size) throws Exception {
+    //This contains the logic about if there is a promote build pending
+    public PromoteResult computePromoteBuildResult(EnvironBean currEnvBean, DeployBean currDeployBean, int size,
+                                                   PromoteBean promoteBean) throws Exception {
+        Date now = new Date();
+        long mustBefore = Long.MAX_VALUE;
+        String schedule = promoteBean.getSchedule();
+        if (!StringUtils.isEmpty(schedule)) {
+            long start_date = 0;
```

Comment:
using high-level object instead of integer to deal with time. If you really want to use integer, name it as something like start_epoch_sec.